### PR TITLE
[fix] check for private var usage in client code instead of server code

### PR DESCRIPTION
**Pull request by benmccann**
Fixes https://github.com/sveltejs/kit/pull/7288

Don't put the module into the graph with `ssrLoadModule` which will add the server-side code, but instead use `ensureEntryFromUrl` leaving the `ssr` parameter `undefined`

### CODE REVIEW

**Review by bluwy — APPROVED** [review]
LGTM. Make sense to move `ssrLoadModule` only after checking the illegal imports.

**Comment by dummdidumm** [comment]
Could we add some comments explaining why we use this other method, and use `resolveSSRModule` only after checking the graph? Someone without context could later come back to this code and think "oh let's move that duplicated code into that resolve function"

**Comment by benmccann** [comment]
ok. I added a couple comments

File: packages/kit/src/exports/vite/dev/index.js [hunk 1]
```diff
@@ -100,27 +100,29 @@ export async function dev(vite, vite_config, svelte_config) {
...
 
-							result.shared = module;
-
 							prevent_illegal_vite_imports(
 								module_node,
 								normalizePath(svelte_config.kit.files.lib),
 								extensions
 							);
+
+							result.shared = await vite.ssrLoadModule(url);
```
Comment (dummdidumm) [inline packages/kit/src/exports/vite/dev/index.js:120]:
```suggestion
							// wait to load the module until after we check for illegal imports
							result.shared = await vite.ssrLoadModule(url);
```

**Review by dummdidumm — APPROVED** [review]

**Pull request merged**